crates/frontend/src/components/toast/mod.rs

```diff
@@ -0,0 +1,72 @@
+use leptos::*;
+
+#[derive(PartialEq, Clone, Debug, strum_macros::Display)]
+#[strum(serialize_all = "lowercase")]
+pub enum AlertType {
+    Info,
+    Success,
+    Warning,
+    Error,
+}
+
+#[derive(Clone, Debug)]
+pub struct Alert {
+    pub id: u64,
+    pub text: String,
+    pub alert_type: AlertType,
+    pub timeout: u64,
+}
+
+impl Alert {
+    pub fn default(id: u64, text: String) -> Self {
```

Comment:
Can we auto generate the ID in default?

Reply:
If u check the id, is generated in `enqueue_alert` fxn, you won't have to give a id of your own most of the time, as for general use-case you can just fire the `enqueue_alert`.